### PR TITLE
[Jest] Migrate `camera.test.js`

### PR DESCRIPTION
**Pull request by wipfli**

### CODE REVIEW

**Comment by wipfli** [comment]
I solved the `0 !== -0` issue with `expect(x).toBeCloseTo(0)`. Hope that is fine.

**Comment by HarelM** [comment]
@astridx defined a different way I believe...?

File: src/ui/camera.test.ts [hunk 1]
```diff
@@ -0,0 +1,1994 @@
...
+import browser from '../util/browser';
+import {fixedLngLat, fixedNum} from '../../test/util/fixed';
+import {setMatchMedia} from '../util/test/util';
+
+beforeEach(() => {
+    setMatchMedia();
+    Object.defineProperty(browser, 'prefersReducedMotion', {value: false});
+});
+
+class TestCamera extends Camera {
```
Comment (HarelM) [inline src/ui/camera.test.ts:13]: CameraMock instead of TestCamera

File: src/ui/camera.test.ts [hunk 2]
```diff
@@ -0,0 +1,1994 @@
...
+
+function attachSimulateFrame(camera) {
+    const queue = new TaskQueue();
+    camera._requestRenderFrame = (cb) => queue.add(cb);
+    camera._cancelRenderFrame = (id) => queue.remove(id);
+    camera.simulateFrame = () => queue.run();
+    return camera;
+}
+
+function createCamera(options) {
```
Comment (HarelM) [inline src/ui/camera.test.ts:32]: Make this optional parameter and remove undefined in all places.

**Review by HarelM — CHANGES_REQUESTED** [review]

**Comment by astridx** [comment]
> @astridx defined a different way I believe...?

At the beginn of the migration I thought that it might be good to have a similar way to handle the zero null. That is why I added it here: https://github.com/maplibre/maplibre-gl-js/issues/458 in point 4.

Now I am not sure if we need a consistent way.

File: src/ui/camera.test.ts (hunk 1 above)
Comment (wipfli) [reply src/ui/camera.test.ts]: Thanks, sounds better indeed.

**Review by HarelM — APPROVED** [review]

File: src/ui/camera.test.ts [hunk 3]
```diff
@@ -0,0 +1,1994 @@
...
+            expect(timeDiff >= min && timeDiff < max).toBeTruthy();
+            done();
+        });
+}
+
+describe('#jumpTo', () => {
+    // Choose initial zoom to avoid center being constrained by mercator latitude limits.
+    const camera = createCamera({zoom: 1});
+
+    test('sets center', done => {
```
Comment (astridx) [inline src/ui/camera.test.ts:62]: I have not yet fully understood the done thing. Why do we need this here?

Comment (HarelM) [reply src/ui/camera.test.ts:62]: You are right, it's not needed here, and in some other tests, I missed that...

**Pull request merged**